test/coinselector.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { Coin } from '../src/wallet/coin';
+import { CoinSelector } from '../src/wallet/coin-selector.ts';
+import { Transaction } from 'bitcoinjs-lib';
+import { validTransactions } from './fixtures/transaction';
+
+describe('CoinSelector', () => {
+    let coins: Coin[];
+    let transaction: Transaction;
+    let coinSelector: CoinSelector;
+    const feeRate: number = 5;
+    let totalBalance: number = 0;
+
+    beforeAll(() => {
+        coinSelector = new CoinSelector(feeRate);
+        // Add Balance
+        coins = [];
+        for (let i = 0; i < 10; i++) {
+            const value = Math.random() * 1000;
+            const coin = new Coin({ value });
+            coins.push(coin);
+            totalBalance += coin.value - coin.estimateSpendingFee(feeRate);
+        }
+    });
+
+    it('should generate cost of change', () => {
+        const costOfChange: number = coinSelector.costOfChange;
+        expect(typeof costOfChange).toBe('number');
+
+        // Calculate transaction output value
+        const txOutValue: number = Math.floor(totalBalance - costOfChange);
+
+        // Create a new transaction
+        transaction = new Transaction();
+        // Add an output with random scriptPubKey: We are only concered about its value
+        transaction.addOutput(
+            Buffer.from(validTransactions[0].raw.outs[0].script, 'hex'),
+            1,
+        );
+
+        // Calculate fees
+        let fees = transaction.virtualSize() * feeRate;
+        fees += 31 * feeRate; // ChangeOutputFee: P2WPKH is 31 B
+        const finalTxOutValue = txOutValue - fees;
+
+        // Set the transaction output value
+        transaction.outs[0].value = finalTxOutValue;
+    });
+
+    it('should add change to fee if it is dust', () => {
```

Comment:
The calculations needed for this test should be done in this test itself. I see that you have a `beforeAll()` and some logic in previous tests, which according to me is misplaced. So I think you should move everything you have in `beforeAll()` and the tx creation part from previous test to this `it()`.